Improve message editing

grahammcculloch opened the pull request:
This PR fixes various small issues relating to editing a message:

* Closes #4197 - you can now use the auto-complete functionality as expected when editing a message.
* Closes #4196 - formatting of usernames, tags etc is re-applied after editing a message
* Closes #4068 - the 'click-outside to save changes' functionality has been replaced by having a 'Save changes' button and a 'Cancel changes' button on the message popup. 

See gif demo below:

![edit message](https://user-images.githubusercontent.com/2925657/85250564-22738b80-b481-11ea-9919-37af055d5f15.gif)

grahammcculloch committed:
Ensure message is re-formatted after editing

Change-type: patch
Signed-off-by: Graham McCulloch <graham@balena.io>

grahammcculloch committed:
Use save and cancel edit butons when editing messages

This makes the save more 'intentional' and makes it obvious how to save (and cancel) your edits. The 'click-outside to save changes' functionality has been removed but you can still press the 'send cmd' to save your changes.

Change-type: minor
Signed-off-by: Graham McCulloch <graham@balena.io>

joshbwlng reviewed (changes requested):
See above comment about possible bug

grahammcculloch committed:
Ensure tokens always re-highlighted after card changes 

This change ensures that whenever the card changes we re-process the message text, re-applying the token highlighting so it stays up-to-date with the message/card.

Change-type: patch
Signed-off-by: Graham McCulloch <graham@balena.io>

grahammcculloch committed:
Mark as read after editing a message

Change-type: patch
Signed-off-by: Graham McCulloch <graham@balena.io>
